### PR TITLE
add iterRow

### PR DESCRIPTION
**Pull request by Penglq**
**What this PR does**:
mysql client add row iterator
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/mysql/client.go [hunk 1]
```diff
@@ -878,6 +878,56 @@ func (conn *BackendConnection) WriteComFieldList(table string, wildcard string)
...
+		return nil, more, warning, err
+	}
+	if colNumber == 0 {
+		// OK packet, means no results. Just use the numbers.
+		return nil, more, warning, nil
+	}
+	iterRow := &IterRow{conn, true, &Row{
+		[]byte{}, &ResultSet{
+			Columns: make([]proto.Field, colNumber),
+		}}}
```
Comment (jjeffcaii) [inline pkg/mysql/client.go:894]: add key

File: pkg/mysql/iterator.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package mysql
```
Comment (jjeffcaii) [inline pkg/mysql/iterator.go:1]: add license

**Review by jjeffcaii — APPROVED** [review]

File: pkg/mysql/iterator.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import "github.com/arana-db/arana/pkg/constants/mysql"
```
Comment (AlexStocks) [inline pkg/mysql/iterator.go:3]: import (
)

File: pkg/mysql/iterator.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+type Iter interface {
+	//HasNextRow() bool
+	Next() (bool, error)
+	Row() *Row
+}
+
+type IterRow struct {
```
Comment (AlexStocks) [inline pkg/mysql/iterator.go:11]: add comment

File: pkg/mysql/iterator.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
+package mysql
+
+import "github.com/arana-db/arana/pkg/constants/mysql"
+
+type Iter interface {
```
Comment (AlexStocks) [inline pkg/mysql/iterator.go:5]: add comment

File: pkg/mysql/client.go [hunk 6]
```diff
@@ -1321,6 +1392,14 @@ func (conn *BackendConnection) PrepareExecuteArgs(query string, args []interface
 	return stmt.execArgs(args)
 }
 
+func (conn *BackendConnection) PrepareQueryArgsIterRow(query string, data []interface{}) (iterRow Iter, warnings uint16, err error) {
```
Comment (AlexStocks) [inline pkg/mysql/client.go:1395]: add comment

File: pkg/mysql/iterator.go [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+func (iterRow *IterRow) Row() *Row {
+	return iterRow.row
+}
+
+//func (iterRow *IterRow) HasNextRow() bool {
+//	return iterRow.hasNext
+//}
+
+func (iterRow *IterRow) Next() (bool, error) {
```
Comment (AlexStocks) [inline pkg/mysql/iterator.go:25]: add comment

**Review by dongzl — APPROVED** [review]

File: pkg/runtime/runtime.go [hunk 8]
```diff
@@ -401,6 +401,30 @@ func (db *AtomDB) CallFieldList(ctx context.Context, table, wildcard string) ([]
...
+	}
+
+	var bc *mysql.BackendConnection
+
+	if bc, err = db.borrowConnection(ctx); err != nil {
+		err = errors.WithStack(err)
+		return
+	}
+
+	defer db.returnConnection(bc)
```
Comment (AlexStocks) [inline pkg/runtime/runtime.go:417]: 合并进一个 defer func() {
xx
xx
}()

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**